src/site/content/en/blog/javascript-and-google-search-io-2019/index.md

```diff
@@ -0,0 +1,619 @@
+
+The default rendering strategy for single-page apps today is client-side rendering. The HTML is loading the JavaScript which is generating the entire content in the browser as it executes. 
+
+<figure class="w-figure">
+  <img src="spa-kittens.png" alt="A code box showing HTML that loads some scripts. A screenshot of a web page on mobile that shows placeholder images while loading the actual content." style="max-width: 400px;">
+  <figcaption class="w-figcaption">
+    The rendering strategy influences the performance and robustness of your web apps.
+  </figcaption>
+</figure> 
+
+The rendering strategy of your website influences performance. It also has an impact on how bots like Googlebot interact with it. Server-side rendering or pre-rendering can improve the performance for users and crawlers. Those strategies execute JavaScript on the server to generate the initial HTML content. The browser can start rendering the HTML as it arrives over the network, making the page load faster.  The [rendering session at I/O](https://www.youtube.com/watch?v=k-A2VfuUROg)  or [the blog post about rendering on the web](https://developers.google.com/web/updates/2019/02/rendering-on-the-web) shows how server-side rendering and hydration can improve the performance and user experience of web apps with more examples of how these strategies are implemented.
+
+If you are looking for a workaround to help crawlers that don't execute JavaScript or cannot make changes to your frontend codebase, consider [dynamic rendering](https://developers.google.com/search/docs/guides/dynamic-rendering) - we also have [a codelab that you can try out](https://codelabs.developers.google.com/codelabs/dynamic-rendering). Unlike server-side rendering or pre-rendering you won't get the user experience or performance benefits as dynamic rendering only serves static HTML to crawlers, so it's a workaround rather than a long-term strategy.
+
+## Test your pages
+
+While most pages generally work fine with Googlebot, we recommend testing your pages regularly to make sure your content is available to Googlebot and there are no problems. The easiest way to do a quick check of a page is the [Mobile-Friendly Test](https://g.co/mobilefriendly). Besides showing you issues with mobile-friendliness, it also gives you a screenshot of the above-the-fold content and the rendered HTML as Googlebot sees it.
+
+<figure class="w-figure">
+  <img src="mobile-friendly-test-rendered-html.png" alt="The mobile-friendly test shows the rendered HTML Googlebot sees after rendering the page" style="max-width: 400px;">
+  <figcaption class="w-figcaption">
+    The mobile-friendly test shows you the rendered HTML Googlebot uses.
+  </figcaption>
+</figure>
+
+You can also find out if there are resource loading issues or JavaScript errors.
+
+<figure class="w-figure">
+  <img src="mobile-friendly-test-js-error.png" alt="The Mobile-Friendly Test shows JavaScript errors and a stack trace." style="max-width: 400px;">
```

Comment:
```suggestion
  <img class="w-screenshot" src="mobile-friendly-test-js-error.png" alt="The Mobile-Friendly Test shows JavaScript errors and a stack trace.">
```

Reply:
Even at full width the text in this image is a little tough to read. Might be worth capturing the image again with the browser font size boosted?